tests/jobserver/test_models.py

```diff
@@ -196,6 +196,335 @@ def test_job_status_unstarted_is_pending():
+        started_at=timezone.now() - timedelta(minutes=1),
+        completed_at=timezone.now(),
+    )
+    job2 = JobFactory(request=request)
+
+    job1.needed_by = job2
+    job1.save()
+
+    assert request.started_at
+    assert not request.runtime
+
+
+@pytest.mark.django_db
+def test_jobrequest_runtime_not_started():
+    request = JobRequestFactory()
+
+    JobFactory(request=request)
+    JobFactory(request=request)
+
+    assert not request.runtime
+
+
+@pytest.mark.django_db
+def test_jobrequest_runtime_success():
+    request = JobRequestFactory()
+
+    start = timezone.now() - timedelta(hours=1)
+
+    job1 = JobFactory(
+        request=request,
+        started_at=start,
+        completed_at=start + timedelta(minutes=1),
+    )
+    job2 = JobFactory(
+        request=request,
+        started_at=start + timedelta(minutes=20),
+        completed_at=start + timedelta(minutes=21),
+    )
+    job3 = JobFactory(
+        request=request,
+        started_at=start + timedelta(minutes=40),
+        completed_at=start + timedelta(minutes=41),
+    )
+    job4 = JobFactory(
+        request=request,
+        started_at=start + timedelta(minutes=50),
+        completed_at=timezone.now(),
+    )
+
+    # set up hierarchy
+    job1.needed_by = job2
+    job1.save()
+
+    job2.needed_by = job4
+    job2.save()
+
+    job3.needed_by = job4
+    job3.save()
+
+    assert request.runtime
+
+
+@pytest.mark.django_db
+def test_jobrequest_started_at_no_jobs():
+    assert not JobRequestFactory().started_at
+
+
+@pytest.mark.django_db
+def test_jobrequest_started_at_success():
+    request = JobRequestFactory()
+
+    job1 = JobFactory(request=request, started_at=timezone.now())
+    job2 = JobFactory(request=request, started_at=timezone.now())
+
+    job1.needed_by = job2
+    job1.save()
+
+    assert request.started_at
+
+
+@pytest.mark.django_db
+def test_jobrequest_status_completed():
+    request = JobRequestFactory()
+
+    job1 = JobFactory(request=request, completed_at=timezone.now())
+    job2 = JobFactory(request=request, completed_at=timezone.now())
+
+    job1.needed_by = job2
+    job1.save()
+
+    assert request.status == "Completed"
+
+
+@pytest.mark.django_db
+def test_jobrequest_status_fall_through():
+    assert JobRequestFactory().status == "Pending"
+
+
+@pytest.mark.django_db
+def test_jobrequest_status_failed():
+    request = JobRequestFactory()
+
+    job1 = JobFactory(request=request, completed_at=timezone.now())
+    job2 = JobFactory(request=request, status_code=3)
+
+    job1.needed_by = job2
+    job1.save()
+
+    assert request.status == "Failed"
+
+
+@pytest.mark.django_db
+def test_jobrequest_status_in_progress():
+    request = JobRequestFactory()
+
+    job1 = JobFactory(request=request, completed_at=timezone.now())
+    job2 = JobFactory(request=request, started_at=timezone.now())
+
+    job1.needed_by = job2
+    job1.save()
+
+    assert request.status == "In Progress"
+
+
+@pytest.mark.django_db
```

Comment:
dupe of `test_jobrequest_status_fall_through`

Reply:
Yep, the catch all in `JobRequest.status` returns `"Pending"` (rather than an implicit `None`) just in case we ever hit that point.